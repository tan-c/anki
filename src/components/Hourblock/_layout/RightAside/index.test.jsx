import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// import { Map } from 'immutable';
import moment from 'moment-timezone';

import { RightAside } from './index';

function setup() {
  const props = {
  };

  return shallow(<RightAside {...props} />);
}

describe('RightAside Unit Tests', () => {
  describe('when normally set up', () => {
    const wrapper = setup();
    wrapper.setState({ todayMoment: moment.tz('2017-01-01', 'Asia/Tokyo') });

    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      // // expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find("aside[data-role='right-aside']")).toHaveLength(1);
    });
  });
});
