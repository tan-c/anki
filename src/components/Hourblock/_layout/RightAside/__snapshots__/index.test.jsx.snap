// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightAside Unit Tests when normally set up renders correctly 1`] = `
<aside
  className="display-right"
  data-role="right-aside"
  id="right-menu"
  style={
    Object {
      "overflow": "auto",
    }
  }
>
  <div
    className="margin-top-10 margin-bottom-10"
  >
    <span
      className="flex-container-row"
    >
      <div
        className="flex-1 false"
        key="1"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        1
      </div>
      <div
        className="flex-1 false"
        key="2"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        2
      </div>
      <div
        className="flex-1 false"
        key="3"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        3
      </div>
      <div
        className="flex-1 false"
        key="4"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        4
      </div>
      <div
        className="flex-1 false"
        key="5"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        5
      </div>
      <div
        className="flex-1 false"
        key="6"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        6
      </div>
      <div
        className="flex-1 false"
        key="7"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        7
      </div>
      <div
        className="flex-1 false"
        key="8"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        8
      </div>
      <div
        className="flex-1 false"
        key="9"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        9
      </div>
      <div
        className="flex-1 color-orange font-400"
        key="10"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        10
      </div>
      <div
        className="flex-1 false"
        key="11"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        11
      </div>
      <div
        className="flex-1 false"
        key="12"
        onClick={[Function]}
        role="button"
        tabIndex="-1"
      >
        12
      </div>
    </span>
    <Connect(MonthlyTasksList)
      activeMonth={9}
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-09-30T15:00:00.000Z",
          "dayOfYearString": "274",
          "isPast": false,
          "isWeekend": false,
          "weekString": "40",
        }
      }
      key="1538319600"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-01T15:00:00.000Z",
          "dayOfYearString": "275",
          "isPast": false,
          "isWeekend": false,
          "weekString": "40",
        }
      }
      key="1538406000"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-02T15:00:00.000Z",
          "dayOfYearString": "276",
          "isPast": false,
          "isWeekend": false,
          "weekString": "40",
        }
      }
      key="1538492400"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-03T15:00:00.000Z",
          "dayOfYearString": "277",
          "isPast": false,
          "isWeekend": false,
          "weekString": "40",
        }
      }
      key="1538578800"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-04T15:00:00.000Z",
          "dayOfYearString": "278",
          "isPast": false,
          "isWeekend": false,
          "weekString": "40",
        }
      }
      key="1538665200"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-05T15:00:00.000Z",
          "dayOfYearString": "279",
          "isPast": false,
          "isWeekend": true,
          "weekString": "40",
        }
      }
      key="1538751600"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-06T15:00:00.000Z",
          "dayOfYearString": "280",
          "isPast": false,
          "isWeekend": true,
          "weekString": "40",
        }
      }
      key="1538838000"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-07T15:00:00.000Z",
          "dayOfYearString": "281",
          "isPast": false,
          "isWeekend": false,
          "weekString": "41",
        }
      }
      key="1538924400"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-08T15:00:00.000Z",
          "dayOfYearString": "282",
          "isPast": false,
          "isWeekend": false,
          "weekString": "41",
        }
      }
      key="1539010800"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-09T15:00:00.000Z",
          "dayOfYearString": "283",
          "isPast": false,
          "isWeekend": false,
          "weekString": "41",
        }
      }
      key="1539097200"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-10T15:00:00.000Z",
          "dayOfYearString": "284",
          "isPast": false,
          "isWeekend": false,
          "weekString": "41",
        }
      }
      key="1539183600"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-11T15:00:00.000Z",
          "dayOfYearString": "285",
          "isPast": false,
          "isWeekend": false,
          "weekString": "41",
        }
      }
      key="1539270000"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-12T15:00:00.000Z",
          "dayOfYearString": "286",
          "isPast": false,
          "isWeekend": true,
          "weekString": "41",
        }
      }
      key="1539356400"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-13T15:00:00.000Z",
          "dayOfYearString": "287",
          "isPast": false,
          "isWeekend": true,
          "weekString": "41",
        }
      }
      key="1539442800"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-14T15:00:00.000Z",
          "dayOfYearString": "288",
          "isPast": false,
          "isWeekend": false,
          "weekString": "42",
        }
      }
      key="1539529200"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-15T15:00:00.000Z",
          "dayOfYearString": "289",
          "isPast": false,
          "isWeekend": false,
          "weekString": "42",
        }
      }
      key="1539615600"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-16T15:00:00.000Z",
          "dayOfYearString": "290",
          "isPast": false,
          "isWeekend": false,
          "weekString": "42",
        }
      }
      key="1539702000"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-17T15:00:00.000Z",
          "dayOfYearString": "291",
          "isPast": false,
          "isWeekend": false,
          "weekString": "42",
        }
      }
      key="1539788400"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-18T15:00:00.000Z",
          "dayOfYearString": "292",
          "isPast": false,
          "isWeekend": false,
          "weekString": "42",
        }
      }
      key="1539874800"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-19T15:00:00.000Z",
          "dayOfYearString": "293",
          "isPast": false,
          "isWeekend": true,
          "weekString": "42",
        }
      }
      key="1539961200"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-20T15:00:00.000Z",
          "dayOfYearString": "294",
          "isPast": false,
          "isWeekend": true,
          "weekString": "42",
        }
      }
      key="1540047600"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-21T15:00:00.000Z",
          "dayOfYearString": "295",
          "isPast": false,
          "isWeekend": false,
          "weekString": "43",
        }
      }
      key="1540134000"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-22T15:00:00.000Z",
          "dayOfYearString": "296",
          "isPast": false,
          "isWeekend": false,
          "weekString": "43",
        }
      }
      key="1540220400"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-23T15:00:00.000Z",
          "dayOfYearString": "297",
          "isPast": false,
          "isWeekend": false,
          "weekString": "43",
        }
      }
      key="1540306800"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-24T15:00:00.000Z",
          "dayOfYearString": "298",
          "isPast": false,
          "isWeekend": false,
          "weekString": "43",
        }
      }
      key="1540393200"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-25T15:00:00.000Z",
          "dayOfYearString": "299",
          "isPast": false,
          "isWeekend": false,
          "weekString": "43",
        }
      }
      key="1540479600"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-26T15:00:00.000Z",
          "dayOfYearString": "300",
          "isPast": false,
          "isWeekend": true,
          "weekString": "43",
        }
      }
      key="1540566000"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-27T15:00:00.000Z",
          "dayOfYearString": "301",
          "isPast": false,
          "isWeekend": true,
          "weekString": "43",
        }
      }
      key="1540652400"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-28T15:00:00.000Z",
          "dayOfYearString": "302",
          "isPast": false,
          "isWeekend": false,
          "weekString": "44",
        }
      }
      key="1540738800"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-29T15:00:00.000Z",
          "dayOfYearString": "303",
          "isPast": false,
          "isWeekend": false,
          "weekString": "44",
        }
      }
      key="1540825200"
    />
    <Connect(DailyTasksList)
      dayVal={
        Object {
          "dayMomentObject": "2018-10-30T15:00:00.000Z",
          "dayOfYearString": "304",
          "isPast": false,
          "isWeekend": false,
          "weekString": "44",
        }
      }
      key="1540911600"
    />
  </div>
</aside>
`;
